core/station/impl/src/services/account.rs

```diff
@@ -510,9 +527,34 @@ impl AccountService {
                     continue;
                 };
 
-                let balance: AccountBalance =
-                    match (&account_asset.balance, balance_considered_fresh) {
-                        (None, _) | (_, false) => {
+                match (&account_asset.balance, balance_considered_fresh) {
+                    (None, _) | (_, false) => {
+                        let balance_update_guard_key = BalanceFetchGuardKey {
+                            account_id: account.id,
+                            asset_id: asset.id,
+                        };
+
+                        let _lock = BALANCE_FETCH_GUARD_STATE.with(|state| {
+                            CallerGuard::new(
+                                state.clone(),
+                                balance_update_guard_key,
+                                Some(time() + Duration::from_secs(60).as_nanos() as u64),
```

Comment:
maybe better to increase this to `5min` in case the subnet would be with high load